src/macros-by-example.adoc

```diff
@@ -711,71 +711,62 @@ Thank you to my Patreon supporters:
 ----
 
 
-[#grouped_parameters]
-=== Grouped Parameters
+=== Argument Groups
 
 The non-rest versions of multi-value parameters can be annoying to invoke, since they require the
 use of `values` or some other template to produce the stream of values.  To streamline invocation,
-a macro can opt-in to special syntax that uses a list as delimiting syntax to group the
-applicable sub-expressions.  This is denoted by wrapping the parameter's type in `[]`:
+there's a special delimiting syntax for an _argument group_ collecting the
+applicable sub-expressions.
 
 [{nrm}]
 ----
 (*macro* prices
-  [(amount [*number*]),      // <1>
+  [(amount *number{asterisk}*),
    (currency *symbol*)]
   (*for* [(amt amount)]
     (price amt currency)))
 ----
 
-<1> Note the use of `[]` around `*number*`.
+Here, the parameter `amount` accepts any number of `*number*`s.
+It's easy to provide exactly one:
 
-This is referred to as a _grouped parameter_, and at invocation it requires a list delimiting its
-_argument group_:
+----
+(:prices 12.99 GBP) ⇒ {amount:12.99, currency:GBP}
+----
+
+To provide a non-singleton stream of values, use an _argument group_.
+Inside an E-expression, a group starts with the digraph `(:` similar to a macro
+invocation, but without a macro reference:
 
 ----
-(:prices [1, 2, 3] GBP) ⇒ {amount:1, currency:GBP}
+(:prices (:) GBP)       ⇒ _void_
+(:prices (: 1) GBP)     ⇒ {amount:1, currency:GBP}
+(:prices (: 1 2 3) GBP) ⇒ {amount:1, currency:GBP}
                           {amount:2, currency:GBP}
                           {amount:3, currency:GBP}
 ----
 
-Within the group, the invocation can have any number of arguments, including macro invocations.
+Within the group, the invocation can have any number of expressions that align
+with the parameter type.
```

Comment:
What are the implications of "that align with the parameter type"? Does this require type checking of expression groups in the macro evaluator? (Seems like this might have some of the trouble of type vs encoding.)

Reply:
It should probably read "base encoding" or something; the terminology is muddled and inconsistent.  Also this would be clarified in the missing section tracked by #303 ... we somehow missed the section defining text E-expression syntax that would be affected by this PR.